hackertracker/HTEventsTableViewController.swift

```diff
@@ -40,37 +40,42 @@ class HTEventsTableViewController: UITableViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        navigationController?.navigationBar.translucent = false
+        navigationController?.navigationBar.isTranslucent = false
         automaticallyAdjustsScrollViewInsets = true
         
-        let delegate : AppDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
+        let delegate : AppDelegate = UIApplication.shared.delegate as! AppDelegate
         let context = delegate.managedObjectContext!
-        let fr:NSFetchRequest = NSFetchRequest(entityName:"Event")
+        let fr:NSFetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName:"Event")
         fr.sortDescriptors = [NSSortDescriptor(key: "begin", ascending: true)]
         fr.returnsObjectsAsFaults = false
         
         for e:eventType in eventTypes {
             fr.predicate = NSPredicate(format: "type = %@", argumentArray: [e.dbName])
-            _ = try! context.executeFetchRequest(fr)
+            _ = try! context.fetch(fr)
             
             var mutableE = e
-            mutableE.count = context.countForFetchRequest(fr, error: nil)
+
+            do {
+                mutableE.count = try context.count(for: fr)
+            } catch {
+                print("failed to retrieve count")
```

Comment:
If you actually somehow got something that wasn't an Integer for a count I'd seriously throw an error and kill the app. You should always get 0 at the very least.